src/persist-client/src/fetch.rs

```diff
@@ -1004,6 +1039,63 @@ impl<T: Timestamp + Codec64> RustType<(ProtoLeasedBatchPart, Arc<Metrics>)> for
     }
 }
 
+/// Format we'll use when decoding a [`Part`].
+#[derive(Debug, Copy, Clone)]
+pub enum PartDecodeFormat {
+    /// Decode from opaque `Codec` data.
+    Row {
+        /// Will also decode the structured data, and validate it matches.
+        validate_structured: bool,
+    },
+}
+
+impl PartDecodeFormat {
+    /// Returns a default value for [`PartDecodeFormat`].
+    pub const fn default() -> Self {
+        // IMPORTANT: By default we will not decode or validate our structured format until it's
+        // more stable.
+        PartDecodeFormat::Row {
+            validate_structured: false,
+        }
+    }
+
+    /// Parses a [`PartDecodeFormat`] from the provided string, falling back to the default if the
+    /// provided value is unrecognized.
+    pub fn from_str(s: &str) -> Self {
+        match s {
+            "row" => PartDecodeFormat::Row {
+                validate_structured: false,
+            },
+            "row_with_validate" => PartDecodeFormat::Row {
+                validate_structured: true,
+            },
+            x => {
+                let default = PartDecodeFormat::default();
+                tracing::error!("Invalid part decode format: '{x}', falling back to {default}");
```

Comment:
we should do something that fails CI here

Reply:
done!